packages/next/apollo.jsx

```diff
@@ -43,20 +46,46 @@ let createApolloClient = (initialState, ctx, apolloCacheConfig, apolloClientConf
         }
     }
 
+    const hasFiles = (data) => {
+        let i = 0
+        let found
+
+        const keys = Object.keys(data)
+
+        while (!found && i < keys.length) {
+            if (data[keys[i]] instanceof File || data[keys[i]] instanceof Blob) {
+                found = true
+            } else {
+                i++
+            }
+        }
+
+        return found
+    }
+
+    const linkPayload = {
+        uri: apolloGraphQLUrl, // Server URL (must be absolute)
+        credentials: 'include',
+        fetchOptions: {
+            mode: 'cors',
+        },
+        fetch: (isOnClientSide && window.fetch) ? window.fetch : fetch,
+        headers: (ctx && ctx.req) ? ctx.req.headers : undefined,  // allow to use client cookies on server side requests
+    }
+
+    const uploadLink = createUploadLink(linkPayload)
+    const batchLink = new BatchHttpLink({
+        ...linkPayload,
+        batchMax: 50,
+        batchInterval: 20,
+    })
+
     // The `ctx` (NextPageContext) will only be present on the server.
     // use it to extract auth headers (ctx.req) or similar.
     return new ApolloClient({
         // connectToDevTools: !Boolean(ctx),
         ssrMode: Boolean(ctx),
-        link: createUploadLink({
-            uri: apolloGraphQLUrl, // Server URL (must be absolute)
-            credentials: 'include',
-            fetchOptions: {
-                mode: 'cors',
-            },
-            fetch: (isOnClientSide && window.fetch) ? window.fetch : fetch,
-            headers: (ctx && ctx.req) ? ctx.req.headers : undefined,  // allow to use client cookies on server side requests
-        }),
+        link: ApolloLink.split(operation => hasFiles(get(operation, 'variables.data', {})), uploadLink, batchLink),
```

Comment:
what if I have a custom mutation with different name? variables.file or smth like that

Reply:
I don't get it. What the behaviour of "different name"? For the second point I agree with you. But if we look from the other side - for custom variables definition blob type search you should recursively search for blobs at all of the keys. So that's might be very complex operation.